src/registries/npm-registry.js

```diff
@@ -346,4 +330,15 @@ export default class NpmRegistry extends Registry {
   getRegistryOrGlobalOption(registry: string, option: string): mixed {
     return this.getRegistryOption(registry, option) || this.getOption(option);
   }
+
+  getAvailableRegistries(): Array<string> {
+    const availableRegistries = super.getAvailableRegistries();
+    if (availableRegistries.indexOf(YARN_REGISTRY) === -1) {
+      availableRegistries.push(YARN_REGISTRY);
+    }
+    if (availableRegistries.indexOf(DEFAULT_REGISTRY) === -1) {
+      availableRegistries.push(DEFAULT_REGISTRY);
+    }
+    return availableRegistries;
+  }
```

Comment:
`DEFAULT_REGISTRY` and `YARN_REGISTRY` are both available registries even if not mentioned in any of the config. Make sure this is reflected as we now rely on this when extracting registry from a request url.

In theory only `DEFAULT_REGISTRY` should suffice, but I wanted to make sure that if someone sets their default registry for some reason to be the yarn registry, things should work the same.